frontend/src/context/AuthContextProvider.tsx

```diff
@@ -65,142 +30,115 @@ export const AuthContextProvider: React.FC = ({ children }) => {
-          '/auth/callback',
-          {
-            body: {
-              token: session.getIdToken().getJwtToken()
-            }
+  const refreshUser = useCallback(async () => {
+    if (!token && process.env.REACT_APP_USE_COGNITO) {
+      const session = await Auth.currentSession();
+      const { token } = await apiPost<{ token: string; user: User }>(
+        '/auth/callback',
+        {
+          body: {
+            token: session.getIdToken().getJwtToken()
           }
-        );
-        await login(token, user);
-      } else {
-        return;
-      }
+        }
+      );
+      setToken(token);
     }
-    const user: User = await apiGet('/users/me');
-    const userCopy: AuthUser = {
-      isRegistered: user.firstName !== '',
-      ...user
-    };
-    localStorage.setItem('user', JSON.stringify(userCopy));
-    setUser(userCopy);
-  };
-
-  const prepareInit = useCallback(async (init: any) => {
-    const { headers, ...rest } = init;
-    return {
-      ...rest,
-      headers: {
-        ...headers,
-        ...baseHeaders,
-        Authorization: localStorage.getItem('token')
-      }
-    };
-  }, []);
+  }, [apiPost, setToken, token]);
 
-  const apiGet = useCallback(
-    async <T extends object = {}>(path: string, init: any = {}) => {
-      try {
-        setLoading(l => l + 1);
-        const options = await prepareInit(init);
-        const result = await API.get('crossfeed', path, options);
-        setLoading(l => l - 1);
-        return result as T;
-      } catch (e) {
-        setLoading(l => l - 1);
-        await handleError(e);
-        throw e;
-      }
-    },
-    [prepareInit, handleError]
-  );
-
-  const apiPost = useCallback(
-    async <T extends object = {}>(path: string, init: any) => {
-      try {
-        setLoading(l => l + 1);
-        const options = await prepareInit(init);
-        const result = await API.post('crossfeed', path, options);
-        setLoading(l => l - 1);
-        return result as T;
-      } catch (e) {
-        setLoading(l => l - 1);
-        await handleError(e);
-        throw e;
-      }
-    },
-    [prepareInit, handleError]
-  );
+  const extendedOrg = useMemo(() => {
+    let current: CurrentOrganization | null =
+      org ?? authUser?.roles[0]?.organization ?? null;
+    if (!current) {
+      return null;
+    }
+    current.userIsAdmin =
+      authUser?.userType === 'globalAdmin' ||
+      authUser?.roles.find(role => role.organization.id === org?.id)?.role ===
+        'admin';
+    return current;
+  }, [org, authUser]);
+
+  const maximumRole = useMemo(() => {
+    if (authUser?.userType === 'globalView') return 'user';
+    return authUser &&
+      authUser.roles &&
+      authUser.roles.find(role => role.role === 'admin')
+      ? 'admin'
+      : 'user';
+  }, [authUser]);
+
+  const touVersion = useMemo(() => `v${currentTermsVersion}-${maximumRole}`, [
+    maximumRole
+  ]);
+
+  const userMustSign = useMemo(() => {
```

Comment:
Would it be possible to factor out these ToU functions (userMustSign, touVersion, and maximumRole) into pure-js functions in a separate `.js` file? This is because we'll need to reuse much of this same code for the backend check (see #316), and it would be easier if we could just copy-and-paste changes from the frontend to the backend to keep the code in sync.

Reply:
separated into `/src/context/userStateUtils.ts`